### ISSUE TITLE
net/http/httputil: NewSingleHostReverseProxy doesn't set RawPath when creating Director

### ISSUE THREAD

**Issue by arun0009** [golang/go#41082]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go1.14 darwin/amd64
</pre>

### Does this issue reproduce with the latest release?
Yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/cxi624/Library/Caches/go-build"
GOENV="/Users/cxi624/Library/Application Support/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOINSECURE=""
GONOPROXY=""
GONOSUMDB=""
GOOS="darwin"
GOPATH="/Users/arung/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/Cellar/go/1.14/libexec"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/Cellar/go/1.14/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/Users/arung/development/src/github.ccom/arung/echo/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build665041636=/tmp/go-build -gno-record-gcc-switches -fno-common"
</pre></details>

### What did you do?

Creating a reverse proxy pass and passing encoded URL path parameter. However, it seems that NewSingleHostReverseProxy doesn't set Raw URL and hence it double escapes the path?

### What did you expect to see?
Pass path parameters **as is** when creating a reverse proxy, but reverse proxy double escapes them. 

However, if the RawPath is set it shouldn't double encode but **NewSingleHostReverseProxy** isn't setting RawPath when creating **Director**

e.g. 

```
package main

import (
	"fmt"
	"net/url"
)

func main() {

	doubleEncodedURL := &url.URL{
		Scheme: "http",
		Host:   "example.com",
		Path:   "/id/EbnfwIoiiXbtr6Ec44sfedeEsjrf0RcXkJneYukTXa%2BIFVla4ZdfRiMzfh%2FEGs7f/events",
	}

	rawEncodedURL := &url.URL{
		Scheme:  "http",
		Host:    "example.com",
		Path:    "/id/EbnfwIoiiXbtr6Ec44sfedeEsjrf0RcXkJneYukTXa+IFVla4ZdfRiMzfh/EGs7f/events",
		RawPath: "/id/EbnfwIoiiXbtr6Ec44sfedeEsjrf0RcXkJneYukTXa%2BIFVla4ZdfRiMzfh%2FEGs7f/events",
	}

```
	fmt.Printf("rawEncodedURL is    : %s\n", rawEncodedURL) //expected NewSingleHostReverseProxy behavior - prints http://example.com/core-events/accounts/EbnfwIoiiXbtr6Ec44sfedeEsjrf0RcXkJneYukTXa%2BIFVla4ZdfRiMzfh%2FEGs7f/events
}

### What did you see instead?
	fmt.Printf("doubleEncodedURL is : %s\n", doubleEncodedURL) //current NewSingleHostReverseProxy behavior - prints http://example.com/core-events/accounts/EbnfwIoiiXbtr6Ec44sfedeEsjrf0RcXkJneYukTXa%252BIFVla4ZdfRiMzfh%252FEGs7f/events

Playground link: https://play.golang.org/p/rOrVzW8ZJCQ

**Comment by cespare** [comment]
@arun0009 it would be helpful if you provided repro code. (Your Go code just prints out some URLs; it doesn't use `httputil.NewSingleHostReverseProxy`.)

From the description this sounds a lot like #35908 which was fixed in 2d323f900df420a29de29cbab949eea08e3d1a61.

You filled out the issue template to say that this problem reproduces with the latest Go release. Are you sure you checked with Go 1.15?

**Comment by arun0009** [comment]
@cespare - I was using go 1.14.4 and updating to go 1.15 fixed the problem. I was just trying to show in my code sample that if you don't set RawPath with encoded value and unescaped Path value, your encoded path can get double escaped. I think in go 1.14.4 it wasn't setting RawPath and was fixed in the commit hash you posted above. Thanks!
